src/modules/facility-fixed-fee/facility-fixed-fee.controller.test.ts

```diff
@@ -0,0 +1,52 @@
+import { GetFacilityFixedFeeGenerator } from '@ukef-test/support/generator/get-facility-fixed-fee-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { FacilityFixedFeeController } from '../facility-fixed-fee/facility-fixed-fee.controller';
+import { FacilityFixedFeeService } from '../facility-fixed-fee/facility-fixed-fee.service';
+
+describe('FacilityFixedFeeController', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const portfolioIdentifier = valueGenerator.string();
+  const facilityIdentifier = valueGenerator.facilityId();
+
+  const { facilityFixedFees: fixedFeesFromService } = new GetFacilityFixedFeeGenerator(valueGenerator, new DateStringTransformations()).generate({
+    numberToGenerate: 2,
+    facilityIdentifier,
+    portfolioIdentifier,
+  });
+
+  let getFacilityFixedFeesService: jest.Mock;
+  let controller: FacilityFixedFeeController;
+
+  beforeEach(() => {
+    const facilityFixedFeeService = new FacilityFixedFeeService(null, null, null);
+    getFacilityFixedFeesService = jest.fn();
+    facilityFixedFeeService.getFixedFeesForFacility = getFacilityFixedFeesService;
+
+    controller = new FacilityFixedFeeController(facilityFixedFeeService);
+  });
+
+  describe('getFixedFeesForFacility', () => {
+    it('returns the fixed fees from the service', async () => {
+      when(getFacilityFixedFeesService).calledWith(facilityIdentifier).mockResolvedValueOnce(fixedFeesFromService);
+
+      const fixedFees = await controller.getFixedFeesForFacility({ facilityIdentifier });
+
+      expect(fixedFees).toStrictEqual(fixedFeesFromService);
+    });
+
+    it('does NOT return unexpected keys returned from the service', async () => {
+      const fixedFeesWithAnUnexpectedKey = fixedFeesFromService.map((fixedFee) => ({
+        ...fixedFee,
+        unexpectedKey: valueGenerator.string(),
+      }));
+      when(getFacilityFixedFeesService).calledWith(facilityIdentifier).mockResolvedValueOnce(fixedFeesWithAnUnexpectedKey);
+
+      const fixedFees = await controller.getFixedFeesForFacility({ facilityIdentifier });
+
+      expect(fixedFees).toStrictEqual(fixedFeesFromService);
+    });
```

Comment:
If you remove the one-to-one mapping in the controller (see comment below), this will also have to be removed.